diff --git a/loader/loader_environment.c b/loader/loader_environment.c
index bdda1a0..cc450a7 100644
--- a/loader/loader_environment.c
+++ b/loader/loader_environment.c
@@ -564,13 +564,14 @@ out:
     return res;
 }
 
-void parse_id_filter_environment_var(const struct loader_instance *inst, const char *env_var_name,
-                                     struct loader_envvar_id_filter *filter_struct) {
+VkResult parse_id_filter_environment_var(const struct loader_instance *inst, const char *env_var_name,
+                                         struct loader_envvar_id_filter *filter_struct) {
+    VkResult res = VK_SUCCESS;
     memset(filter_struct, 0, sizeof(struct loader_envvar_id_filter));
     char *parsing_string = NULL;
     char *env_var_value = loader_secure_getenv(env_var_name, inst);
     if (NULL == env_var_value) {
-        return;
+        return VK_SUCCESS;
     }
     const size_t env_var_len = strlen(env_var_value);
     if (env_var_len == 0) {
@@ -589,6 +590,10 @@ void parse_id_filter_environment_var(const struct loader_instance *inst, const c
 
     filter_struct->filters = (struct loader_envvar_id_filter_value *)loader_instance_heap_alloc(
         inst, (num_commas + 1) * sizeof(struct loader_envvar_id_filter_value), VK_SYSTEM_ALLOCATION_SCOPE_COMMAND);
+    if (NULL == filter_struct->filters) {
+        res = VK_ERROR_OUT_OF_HOST_MEMORY;
+        goto out;
+    }
     filter_struct->count = 0;
     char *context = NULL;
     char *token = thread_safe_strtok(parsing_string, ",", &context);
@@ -612,6 +617,7 @@ void parse_id_filter_environment_var(const struct loader_instance *inst, const c
 out:
 
     loader_free_getenv(env_var_value, inst);
+    return res;
 }
 
 bool check_id_matches_filter_environment_var(const uint32_t id, const struct loader_envvar_id_filter *filter_struct) {
diff --git a/loader/loader_environment.h b/loader/loader_environment.h
index b339a53..2bc1c46 100644
--- a/loader/loader_environment.h
+++ b/loader/loader_environment.h
@@ -55,7 +55,7 @@ VkResult loader_add_environment_layers(struct loader_instance *inst, const char
                                        struct loader_pointer_layer_list *expanded_target_list,
                                        const struct loader_layer_list *source_list);
 
-void parse_id_filter_environment_var(const struct loader_instance *inst, const char *env_var_name,
-                                     struct loader_envvar_id_filter *filter_struct);
+VkResult parse_id_filter_environment_var(const struct loader_instance *inst, const char *env_var_name,
+                                         struct loader_envvar_id_filter *filter_struct);
 bool check_id_matches_filter_environment_var(const uint32_t id, const struct loader_envvar_id_filter *filter_struct);
 void free_id_filters(const struct loader_instance *inst, struct loader_envvar_id_filter *filter_struct);
diff --git a/loader/trampoline.c b/loader/trampoline.c
index 5fe35d2..fd601dd 100644
--- a/loader/trampoline.c
+++ b/loader/trampoline.c
@@ -895,9 +895,12 @@ LOADER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkEnumeratePhysicalDevices(VkInstan
         goto out;
     }
 
-    parse_id_filter_environment_var(inst, VK_DEVICE_ID_FILTER_ENV_VAR, &device_id_filter);
-    parse_id_filter_environment_var(inst, VK_VENDOR_ID_FILTER_ENV_VAR, &vendor_id_filter);
-    parse_id_filter_environment_var(inst, VK_DRIVER_ID_FILTER_ENV_VAR, &driver_id_filter);
+    res = parse_id_filter_environment_var(inst, VK_DEVICE_ID_FILTER_ENV_VAR, &device_id_filter);
+    if (res != VK_SUCCESS) goto out;
+    res = parse_id_filter_environment_var(inst, VK_VENDOR_ID_FILTER_ENV_VAR, &vendor_id_filter);
+    if (res != VK_SUCCESS) goto out;
+    res = parse_id_filter_environment_var(inst, VK_DRIVER_ID_FILTER_ENV_VAR, &driver_id_filter);
+    if (res != VK_SUCCESS) goto out;
 
     // Call down the chain to get the physical device info
     if ((0 == device_id_filter.count) && (0 == vendor_id_filter.count) && (0 == driver_id_filter.count)) {
@@ -2658,9 +2661,12 @@ LOADER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkEnumeratePhysicalDeviceGroups(
         goto out;
     }
 
-    parse_id_filter_environment_var(inst, VK_DEVICE_ID_FILTER_ENV_VAR, &device_id_filter);
-    parse_id_filter_environment_var(inst, VK_VENDOR_ID_FILTER_ENV_VAR, &vendor_id_filter);
-    parse_id_filter_environment_var(inst, VK_DRIVER_ID_FILTER_ENV_VAR, &driver_id_filter);
+    res = parse_id_filter_environment_var(inst, VK_DEVICE_ID_FILTER_ENV_VAR, &device_id_filter);
+    if (res != VK_SUCCESS) goto out;
+    res = parse_id_filter_environment_var(inst, VK_VENDOR_ID_FILTER_ENV_VAR, &vendor_id_filter);
+    if (res != VK_SUCCESS) goto out;
+    res = parse_id_filter_environment_var(inst, VK_DRIVER_ID_FILTER_ENV_VAR, &driver_id_filter);
+    if (res != VK_SUCCESS) goto out;
 
     // Call down the chain to get the physical device group info.
     if ((0 == device_id_filter.count) && (0 == vendor_id_filter.count) && (0 == driver_id_filter.count)) {
